app/src/main/java/ch/epfl/sdp/blindly/main_screen/audio_player/AudioPlayerFragment.kt

```diff
@@ -0,0 +1,121 @@
+package ch.epfl.sdp.blindly.main_screen.audio_player
+
+import android.content.Intent
+import android.media.MediaPlayer
+import android.os.Build
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.annotation.RequiresApi
+import androidx.appcompat.widget.AppCompatImageButton
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.commit
+import ch.epfl.sdp.blindly.R
+import ch.epfl.sdp.blindly.recording.AudioRecord
+import ch.epfl.sdp.blindly.recording.BlindlyMediaPlayer
+import ch.epfl.sdp.blindly.recording.PRESENTATION_AUDIO_NAME
+import ch.epfl.sdp.blindly.recording.RecordingActivity
+import java.io.File
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [AudioPlayerFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+
+class AudioPlayerFragment : Fragment() {
+    private val blindlyMediaPlayer = BlindlyMediaPlayer()
+    private lateinit var audioRecord: AudioRecord
+    private lateinit var recordName: TextView
+    private lateinit var recordDuration: TextView
+    private lateinit var playBar: SeekBar
+    private lateinit var remainingTimer: Chronometer
+    private lateinit var playTimer: Chronometer
+    private lateinit var playPauseButton: AppCompatImageButton
+
+    /**
+     * Instantiates the mediaPlayer
+     *
+     * @param inflater
+     * @param container
+     * @param savedInstanceState
+     * @return the AudioPlayerFragment view
+     */
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view = inflater.inflate(R.layout.fragment_audio_player, container, false)
+
+        val file = File("${context?.filesDir?.absolutePath}/$PRESENTATION_AUDIO_NAME")
+        //TODO Can we obtain the duration from a File?

```

Comment:
I think there's a way if you play with metadata, but I'm not sure. At least it's possible when you upload on the Firebase storage xD

Reply:
I completely removed it, since I didn't find a way to get it.... So I'll also remove the TODO x)